lib/fs/walkfs.go

```diff
@@ -38,7 +38,13 @@ func NewWalkFilesystem(next Filesystem) Filesystem {
 
 // walk recursively descends path, calling walkFn.
 func (f *walkFilesystem) walk(path string, info FileInfo, walkFn WalkFunc) error {
-	err := walkFn(path, info, nil)
+	var err error
+	path, err = canonicalize(path)
```

Comment:
You can just `path, err :=` here (and in the other use of canonicalize above) but I don't particularly care either way.